package/libsolv.spec.in

```diff
@@ -15,122 +15,97 @@
-%bcond_without disable_shared
-%bcond_without perl_binding
-%bcond_without python_binding
-%bcond_without python3_binding
-%bcond_without ruby_binding
-%bcond_with zypp
 
+%define libname libsolv@LIBSOLV_SOVERSION@
 %if 0%{?leap_version} >= 420300 || 0%{?sle_version} >= 120300 || 0%{?suse_version} >= 1330 || !0%{?suse_version}
 %bcond_without bz2
 %bcond_without xz
 %else
 %bcond_with bz2
 %bcond_with xz
 %endif
-
 %if 0%{?fedora} || 0%{?rhel} >= 7 || 0%{?mageia} >= 6 || 0%{?suse_version} >= 1330
 %bcond_without richdeps
 %else
 %bcond_with richdeps
 %endif
-
-%if 0%{?mandriva_version}
-# force this version on mandriva
-BuildRequires:  libneon0.26-devel
+%bcond_without enable_static
+%bcond_without disable_shared
+%bcond_without perl
+%bcond_without python2
+%bcond_without python3
+%bcond_without ruby
+%bcond_with zypp
+Name:           libsolv
+Version:        @VERSION@
+Release:        0
+Summary:        Package dependency solver using a satisfiability algorithm
+License:        BSD-3-Clause
+Group:          Development/Libraries/C and C++
+Url:            https://github.com/openSUSE/libsolv
+Source:         libsolv-%{version}.tar.bz2
+BuildRequires:  cmake
+BuildRequires:  fdupes
+BuildRequires:  gcc-c++
+BuildRequires:  pkgconfig
+BuildRequires:  pkgconfig(libxml-2.0)
+BuildRequires:  pkgconfig(rpm)
+BuildRequires:  pkgconfig(zlib)
+%if %{with ruby}
+BuildRequires:  ruby
+BuildRequires:  ruby-devel
+BuildRequires:  swig
 %endif
 %if 0%{?fedora} || 0%{?rhel} >= 6 || 0%{?mageia}
 BuildRequires:  db-devel
 %endif
-BuildRequires:  libxml2-devel
-%if 0%{?suse_version} && 0%{?suse_version} < 1100
-BuildRequires:  graphviz
-%endif
-%if 0%{?suse_version} > 1020 || 0%{?fedora} || 0%{?mageia}
-BuildRequires:  fdupes
-%endif
-BuildRequires:  cmake
-BuildRequires:  gcc-c++
-BuildRequires:  rpm-devel
-BuildRequires:  zlib-devel
-
-%if %{with perl_binding}
+%if %{with perl}
 BuildRequires:  perl
+BuildRequires:  swig
 %if 0%{?fedora} || 0%{?rhel} >= 6 || 0%{?mageia}
 BuildRequires:  perl-devel
 %endif
-BuildRequires:  swig
 %endif
-
-%if %{with ruby_binding}
-%global ruby_vendorarch %(ruby  -r rbconfig -e "puts RbConfig::CONFIG['vendorarchdir'].nil? ? RbConfig::CONFIG['sitearchdir'] : RbConfig::CONFIG['vendorarchdir']")
-BuildRequires:  ruby
-BuildRequires:  ruby-devel
+%if %{with python2}
 BuildRequires:  swig
+BuildRequires:  pkgconfig(python2)
 %endif
-
-%if %{with python_binding}
-%global python_sitearch %(python -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(True))")
-BuildRequires:  python-devel
+%if %{with python3}
 BuildRequires:  swig
+BuildRequires:  pkgconfig(python3)
 %endif
-
-%if %{with python3_binding}
-%global python3_sitearch %(python3 -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(True))")
-BuildRequires:  python3-devel
-%endif
-
 %if %{with bz2}
-%if 0%{?suse_version}
-BuildRequires:  libbz2-devel
-%else
-BuildRequires:  bzip2-devel
-%endif
+BuildRequires:  pkgconfig(bzip2)
```

Comment:
Are you sure Fedora, Mageia, openSUSE, and RHEL/CentOS ship the pkgconfig file? It's not actually part of the bzip2 software.

I know that at least Fedora and Mageia do...

Reply:
Some older distros should not have it, but they can't read pkgconfig deps either...